Site Editor: Always use full screen mode

creativecoder opened the pull request:
## Description

This change addresses https://github.com/WordPress/gutenberg/issues/26125 and https://github.com/WordPress/gutenberg/issues/29339 by always using full screen mode in the site editor.

- Removes the option to turn off full screen mode
- Displays the navigation toggle ("W" menu) for all screen sizes (rather than hiding it on smaller screens)

## How has this been tested?

- Manually tested the site editor with different screen sizes to verify the desired behavior

## Screenshots

<img width="800" alt="Screen Shot 2021-03-02 at 16 36 03" src="https://user-images.githubusercontent.com/1699996/109726735-d2ed6f80-7b78-11eb-8eef-dde5c2f0ab3a.png">

| Menu closed | Menu open |
| - | - |
| <img width="380" alt="Screen Shot 2021-03-02 at 16 37 45" src="https://user-images.githubusercontent.com/1699996/109726749-db45aa80-7b78-11eb-9f1e-41a24ec4628a.png"> | <img width="381" alt="Screen Shot 2021-03-02 at 16 38 05" src="https://user-images.githubusercontent.com/1699996/109726750-db45aa80-7b78-11eb-82a4-923ce1283b04.png"> |

## Types of changes

Removes a UI setting in the site editor.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] ~~I've included developer documentation if appropriate.~~ (n/a) <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] ~~I've updated all React Native files affected by any refactorings/renamings in this PR.~~ (n/a) <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/native-mobile.md -->

creativecoder committed:
Site Editor: Always use full screen mode

creativecoder committed:
Restore style formatting

Addison-Stavlo reviewed (commented):
These changes look pretty reasonable to me.  I wonder if anyone else involved with the Nav Panel might notice something we might have overlooked, but it seems to work well!

Another thought (although potentially as a follow up), is maybe we should be hiding the top-bar that is shown in these smaller screen widths in the site editor:

![Screen Shot 2021-03-03 at 1 55 37 PM](https://user-images.githubusercontent.com/28742426/109857495-c1fc3700-7c28-11eb-8ac0-f11e9ed8b9af.png)

With the Nav Panel available, this top bar really doesn't bring much to the user. 🤔

creativecoder committed:
Removes superfluous true prop

creativecoder commented on packages/edit-site/src/components/editor/index.js:
```diff
@@ -169,7 +166,7 @@ function Editor( { initialSettings } ) {
 		<>
 			<URLQueryController />
-			<FullscreenMode isActive={ isFullscreenActive } />
+			<FullscreenMode isActive={ true } />
```
Good point! Updated in https://github.com/WordPress/gutenberg/pull/29489/commits/da8c5ea398596c374ff586809d63bebc10b354de

creativecoder commented:
Note that I ran into a keyboard focus issue while testing this (not caused by these changes). Opened https://github.com/WordPress/gutenberg/issues/29536 for tracking.

david-szabo97 commented:
**Blocker:**
Just noticed in the Menu Closed screenshot that the inserter button is missing. Spun up your PR and found out the inserter button goes behind the W button. I think we need to remove the media query from here:

https://github.com/WordPress/gutenberg/blob/10ee7eec01935d56562fecb4217c59b6f4881ecd/packages/edit-site/src/components/header/style.scss#L11-L20

**Non-blocker:** (follow-up?)
I wonder if we could somehow force a `min-width` on the editor to avoid this ugly header and editor content:
![image](https://user-images.githubusercontent.com/2256104/109940733-b2392d00-7cd2-11eb-9f88-3d8ed2862e9a.png)

Or how about making the sidebar full width below 400-500px? 🤔

jameskoster commented:
Closing the sidebar upon selection of a menu item feels loads better to me 🎉

> maybe we should be hiding the top-bar that is shown in these smaller screen widths in the site editor

I'd agree with that. We hide the admin bar on desktop view, probably makes sense to hide it on mobile as well. +1 to doing that in a follow-up though, in case there is some context missing as to why it is not currently hidden.

creativecoder committed:
Fix missing inserter in header on small screens

creativecoder commented:
> Blocker:
> Just noticed in the Menu Closed screenshot that the inserter button is missing.

@david-szabo97 Thanks for catching this! Addressed in https://github.com/WordPress/gutenberg/pull/29489/commits/7338f1abb85116b6ebd15fbeb21d1863502b10e3 by removing the media query, as suggested.

> I wonder if we could somehow force a min-width on the editor to avoid this ugly header and editor content
> Or how about making the sidebar full width below 400-500px? 🤔

I think that's a good idea. I'd like to push this to a follow-up PR, as it also brings into question some additional concerns... for example if the navigation panel is full screen, the [lack of an obvious close button](https://github.com/WordPress/gutenberg/issues/29529) becomes more of an problem.

david-szabo97 commented:
Something is going on with the E2E tests... Can you try a rebase? 🙏 Thanks!

I just noticed that the block toolbar overlaps the browsing sidebar on medium screen sizes. Let's fix that in a follow-up. 
https://github.com/WordPress/gutenberg/issues/29615

creativecoder commented:
> Something is going on with the E2E tests... Can you try a rebase? 🙏 Thanks!

Thanks for pointing that out, @david-szabo97 ! I tried rebasing twice so far today, but some tests are still failing.

All failures are related to the block inserter not being open to insert a block. So far I'm unable to replicate the failure locally, so I'm not sure what's preventing the inserter from opening.

david-szabo97 commented:
I can confirm after commenting out these two lines, tests are passing.

https://github.com/creativecoder/gutenberg/blob/7580d5aca19ea7317d6d958aadf824ab9980cef6/packages/edit-site/src/components/navigation-sidebar/navigation-panel/content-navigation-item.js#L51

and 

https://github.com/creativecoder/gutenberg/blob/7580d5aca19ea7317d6d958aadf824ab9980cef6/packages/edit-site/src/components/navigation-sidebar/navigation-panel/template-navigation-item.js#L49

david-szabo97 commented:
The inserter button goes behind the W button for a few milliseconds. Puppeteer tries to click on it but it's not visible in the viewport. Instead of clicking the inserter button it clicks on the W button. That's why in the screenshot the browsing sidebar is still open.

Opened a PR that will fix the problem above.
https://github.com/WordPress/gutenberg/pull/29762

david-szabo97 commented:
I think we can remove all instances of `await navigationPanel.close();` after `await navigationPanel.clickItemByText` calls. Since we are now closing the browsing sidebar automatically.

https://github.com/WordPress/gutenberg/blob/c54caa2f71e2a7804b85910026585606c8670403/packages/e2e-tests/specs/experiments/multi-entity-saving.test.js#L190
https://github.com/WordPress/gutenberg/blob/9bf17a6c0a8a7c6f9033862c7d6a59acda66582c/packages/e2e-tests/specs/experiments/settings-sidebar.test.js#L74
https://github.com/WordPress/gutenberg/blob/d605d15305293e7a8f83919a022d3f2c83a5d7c6/packages/e2e-tests/specs/experiments/template-part.test.js#L69

creativecoder commented:
@david-szabo97 Thanks so much for helping me figure out what's going on with the e2e tests, and for your fix in #29762!

There's enough complexity going on with the e2e tests here that I'm going with your idea of using a separate PR: I've removed the change that closes the navigation panel after selecting a template and updated this PR description. I'll open a separate PR with that feature + the associated e2e test changes.

vindl reviewed (commented):
This is working nicely for me! 

Should we be displaying the default `wpadminbar` at the top on smaller screens (<782px)? I'm mostly using Customizer here for comparison, which is another editor that forces fullscreen. In that case it's not shown and I'm thinking we should hide it in the case of Site Editor too.

I also noticed that exactly at 782px width we end up in this weird state where admin bar takes up the space but it's not shown:

<img width="618" alt="Screenshot 2021-03-11 at 21 51 18" src="https://user-images.githubusercontent.com/1182160/110854156-d25d8300-82b4-11eb-98dc-47d9b28a4e60.png">

creativecoder commented:
> Should we be displaying the default wpadminbar at the top on smaller screens (<782px)?

@vindl I think the answer is no, to be consistent with how full screen mode works. (@Addison-Stavlo made the same suggestion [in an earlier comment](https://github.com/WordPress/gutenberg/pull/29489#pullrequestreview-603242952)).

I'm tracking that in https://github.com/WordPress/gutenberg/issues/29805

creativecoder commented:
This is now passing CI and should be ready for a formal review.

I've moved the feature to auto-close the sidebar after selection to https://github.com/WordPress/gutenberg/pull/29806

jameskoster reviewed (approved):
Approving from the design side :)

vindl commented:
> @vindl I think the answer is no, to be consistent with how full screen mode works. (@Addison-Stavlo made the same suggestion in an earlier comment).
> I'm tracking that in #29805

Great, I think it makes sense to address this in a follow-up since it's a separate decision. This is looking good to me now! :shipit:
